File: tracker/src/test/java/org/matomo/sdk/tools/ChecksumTest.java
package org.matomo.sdk.tools;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import java.io.File;

import testhelpers.BaseTest;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

@RunWith(MockitoJUnitRunner.class)
public class ChecksumTest extends BaseTest {

    @Test
    public void testgetMD5Checksum() throws Exception {
        String md5 = Checksum.getMD5Checksum("foo");
        assertEquals(md5, "ACBD18DB4CC2F85CEDEF654FCCC4A4D8");
    }

    @Test
    public void testHex() {
        assertNull(Checksum.getHex(null));
    }

    @Test
    public void testgetMD5ChecksumDir() throws Exception {
        File directory = new File(".", "");
        String md5 = Checksum.getMD5Checksum(directory);
        assertNull(md5);
    }

}

File: tracker/src/test/java/org/matomo/sdk/tools/ChecksumTest.kt
package org.matomo.sdk.tools

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner
import testhelpers.BaseTest
import java.io.File

@RunWith(MockitoJUnitRunner::class)
class ChecksumTest : BaseTest() {
    @Test
    @Throws(Exception::class)
    fun testgetMD5Checksum() {
        val md5 = Checksum.getMD5Checksum("foo")
        Assert.assertEquals(md5, "ACBD18DB4CC2F85CEDEF654FCCC4A4D8")
    }

    @Test
    fun testHex() {
        Assert.assertNull(Checksum.getHex(null))
    }

    @Test
    @Throws(Exception::class)
    fun testgetMD5ChecksumDir() {
        val directory = File(".", "")
        val md5 = Checksum.getMD5Checksum(directory)
        Assert.assertNull(md5)
    }
}
